File: 2610. Convert an Array Into a 2D Array With Conditions.java
class Solution {
    public List<List<Integer>> findMatrix(int[] nums) {      
        Map<Integer,Integer> mp=new HashMap<>();
        Set<Integer> set=new HashSet<>();
        for(int i:nums)
        {
            set.add(i);
            mp.put(i,mp.getOrDefault(i,0)+1);
        }
        List<List<Integer>> answer=new ArrayList<>();
        while(mp.size()!=0)
        {
            List<Integer> list=new ArrayList<>();
            for(int i:set) {

                if(mp.containsKey(i))
                {
                    int val = mp.get(i);
                    if (val > 0) {
                        list.add(i);
                        mp.put(i, val - 1);
                    } else {
                        mp.remove(i);
                    }
                }
            }
            answer.add(list);
        }
        answer.remove(answer.size()-1);
        return answer;      
    }
}

File: 2610. Convert an Array Into a 2D Array With Conditions.cpp
class Solution {
public:
    vector<vector<int>> findMatrix(vector<int>& nums) {
        vector<vector<int>>ans;
        unordered_map<int, int>mp;
        for(auto i:nums)
            mp[i]++;
        while(true)
        {
            vector<int>v;
            for(auto [key,value]:mp)
            {
                if(value)
                {
                    v.push_back(key);
                    mp[key]--;
                }
            }
            if(!v.size())
                break;
            ans.push_back(v);
        }
        return ans;
    }
};
